go-controller/pkg/ovn/egressip.go

```diff
@@ -1609,18 +1259,30 @@ func (oc *DefaultNetworkController) generateCacheForEgressIP() (map[string]egres
 	for _, egressIP := range egressIPs {
 		egressIPCache[egressIP.Name] = egressIPCacheEntry{
 			egressPods:       make(map[string]sets.Set[string]),
-			gatewayRouterIPs: sets.New[string](),
+			gatewayRouterIPs: sets.New[string](), // can be transit switchIPs for interconnect multizone setup
 			egressIPs:        map[string]string{},
 		}
+		isEgressNodeLocal := false
 		for _, status := range egressIP.Status.Items {
+			var nextHopIP string
 			isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
-			gatewayRouterIP, err := oc.eIPC.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-			if err != nil {
-				klog.Errorf("Unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
-				continue
+			_, ok := oc.localZoneNodes.Load(status.Node)
+			if !config.OVNKubernetesFeature.EnableInterconnect || ok {
```

Comment:
if localZones was always accurate, we could just:
```
if _, isLocalZoneNode := oc.localZoneNodes.Load(status.Node) ; isLocalZoneNode {
...
}
```
wdyt?

Reply:
this is related to the previous comment right ? -> https://github.com/ovn-org/ovn-kubernetes/pull/3386/files#r1186521805 ? What you mean here is to remove the bool check for whether the feature is enabled or not?

Reply:
yup, sorry for the repeat. I think you answered that above.